src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/EventListener.java

```diff
@@ -164,25 +166,39 @@ protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent
      * @param event   the event.
      * @param project the project to build.
      */
-    protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent event, AbstractProject project) {
+    protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent event, Job project) {
         BadgeAction badgeAction = new BadgeAction(event);
         //during low traffic we still don't want to spam Gerrit, 3 is a nice number, isn't it?
         int projectbuildDelay = t.getBuildScheduleDelay();
         if (cause instanceof GerritUserCause) {
             // it's a manual trigger, no need for a quiet period
             projectbuildDelay = 0;
-        } else if (project.getHasCustomQuietPeriod()
-                && project.getQuietPeriod() > projectbuildDelay) {
-            projectbuildDelay = project.getQuietPeriod();
+        } else if (project instanceof AbstractProject) {
```

Comment:
So, if I understand this right, the old behavior was something like:

2. If we're manual trigger, don't delay at all
3. If the job has a custom quiet period (ie: not default) AND that period is larger than the ScheduleDelay, use this
4. Otherwise, use the ScheduleDelay 

So what's wrong with using the quietPeriodDelay even if it's not customized? Why can't we also check if we're above the default?

Is it so that GT's custom delay overrides the global default, but not per-jobs?

I'd rather use the longest of all configured delays, not the shortest. Ie: GT's delay only overrides global if the global is shorter?

We could also use only GT's trigger period, without worrying about per-job settings?

Thoughts?